lib/Entity/Media.php

```diff
@@ -901,28 +901,33 @@ private function assessDimensions()
             // Media released set to 0 for large size images
             // if image size is greater than 8000 X 8000 then we flag that image as too big
             if ($imgWidth > $resizeLimit || $imgHeight > $resizeLimit) {
```

Comment:
What you've got here will definitely work - but I think we might be able to make it more DRY

```
if ($resizeLimit > 0 && ($imgWidth > $resizeLimit || $imgHeight > $resizeLimit)) {
```

Then have another elseif:

```
elseif ($resizeLimit <= 0) {
   // the current imgWidth > $imgHeight clauses...
}
```
```